lib/elixir/test/elixir/exception_test.exs

```diff
@@ -335,6 +335,33 @@ defmodule ExceptionTest do
     assert %ArgumentError{message: "unexpected comté"} |> message == "unexpected comté"
   end
 
+  describe "Process.RegisterError message" do
+
+    test "registering dead process" do
+      error = assert_raise ArgumentError, fn ->
+        pid = spawn(fn() -> 1 end)
+        Process.exit(pid, :kill)
+        Process.register(pid, :my_process)
+      end
+
+      assert String.contains?(error.message, "is no longer alive")
+    end
+
+    test "registering duplicate name" do
+      error = assert_raise ArgumentError, fn ->
+        Process.register(self(), :my_process)
+        Process.register(self(), :my_process)
+      end
+
+      if is_pid(Process.whereis(:my_process)) do
+        Process.unregister(:my_process)
+      end
```

Comment:
There shouldn't be any need for this. The process will die after the test finishes.